SerialPrograms/Source/PokemonHome/Programs/PokemonHome_BoxSorting.cpp

```diff
@@ -618,14 +617,10 @@ void BoxSorting::program(SingleSwitchProgramEnvironment& env, BotBaseContext& co
 
                     if(boxes_data[get_index(box_nb, row, column)].has_value()){
                         screen = env.console.video().snapshot();
-                        ImageRGB32 image = to_blackwhite_rgb32_range(
-                            extract_box_reference(screen, national_dex_number_box),
-                            0xff808080, 0xffffffff, true
-                            );
-
-                        int national_dex_number = OCR::read_number(env.console, image);
-                        if (national_dex_number == -1){
-                            dump_image(env.console, ProgramInfo(), "ReadSummary", screen);
+
+                        int national_dex_number = OCR::read_number_waterfill(env.console, extract_box_reference(screen, national_dex_number_box), 0xff808080, 0xffffffff);
+                        if (national_dex_number < 0 || national_dex_number > 1025) {

```

Comment:
it will prevent this to run with next game but we have time and it might not even be on the switch. But that way, we can catch misread for current pokemon